gov2/dynamodb/actions/partiql.go

```diff
@@ -0,0 +1,275 @@
+		log.Printf("Couldn't get info about %v. Here's why: %v\n", title, err)
+	} else {
+		err = attributevalue.UnmarshalMap(response.Items[0], &movie)
+		if err != nil {
+			log.Printf("Couldn't unmarshal response. Here's why: %v\n", err)
+		}
+	}
+	return movie, err
+}
+
+// snippet-end:[gov2.dynamodb.ExecuteStatement.Select]
+
+// snippet-start:[gov2.dynamodb.ExecuteStatement.Select.Projected]
+
+// GetAllMovies runs a PartiQL SELECT statement to get all movies from the DynamoDB table.
+// The results are projected to return only the title and rating of each movie.
+func (runner PartiQLRunner) GetAllMovies() ([]map[string]interface{}, error) {
+	var output []map[string]interface{}
+	response, err := runner.DynamoDbClient.ExecuteStatement(context.TODO(), &dynamodb.ExecuteStatementInput{
+		Statement: aws.String(
+			fmt.Sprintf("SELECT title, info.rating FROM \"%v\"", runner.TableName)),
+	})
+	if err != nil {
+		log.Printf("Couldn't get movies. Here's why: %v\n", err)
+	} else {
+		err = attributevalue.UnmarshalListOfMaps(response.Items, &output)
+		if err != nil {
+			log.Printf("Couldn't unmarshal response. Here's why: %v\n", err)
+		}
+	}
+	return output, err
+}
+
+// snippet-end:[gov2.dynamodb.ExecuteStatement.Select.Projected]
+
+// snippet-start:[gov2.dynamodb.ExecuteStatement.Update]
+
+// UpdateMovie runs a PartiQL UPDATE statement to update the rating of a movie that
+// already exists in the DynamoDB table.
+func (runner PartiQLRunner) UpdateMovie(movie Movie, rating float64) error {
+	params, err := attributevalue.MarshalList([]interface{}{rating, movie.Title, movie.Year})
+	if err != nil {
+		panic(err)
+	}
+	_, err = runner.DynamoDbClient.ExecuteStatement(context.TODO(), &dynamodb.ExecuteStatementInput{
+		Statement: aws.String(
+			fmt.Sprintf("UPDATE \"%v\" SET info.rating=? WHERE title=? AND year=?",
+				runner.TableName)),
+		Parameters: params,
+	})
+	if err != nil {
+		log.Printf("Couldn't update movie %v. Here's why: %v\n", movie.Title, err)
+	}
+	return err
+}
+
+// snippet-end:[gov2.dynamodb.ExecuteStatement.Update]
+
+// snippet-start:[gov2.dynamodb.ExecuteStatement.Delete]
+
+// DeleteMovie runs a PartiQL DELETE statement to remove a movie from the DynamoDB table.
+func (runner PartiQLRunner) DeleteMovie(movie Movie) error {
+	params, err := attributevalue.MarshalList([]interface{}{movie.Title, movie.Year})
+	if err != nil {
+		panic(err)
+	}
+	_, err = runner.DynamoDbClient.ExecuteStatement(context.TODO(), &dynamodb.ExecuteStatementInput{
+		Statement: aws.String(
+			fmt.Sprintf("DELETE FROM \"%v\" WHERE title=? AND year=?",
+				runner.TableName)),
+		Parameters: params,
+	})
+	if err != nil {
+		log.Printf("Couldn't delete %v from the table. Here's why: %v\n", movie.Title, err)
+	}
+	return err
+}
+
+// snippet-end:[gov2.dynamodb.ExecuteStatement.Delete]
+
+// snippet-start:[gov2.dynamodb.BatchExecuteStatement.Insert]
+
+// AddMovieBatch runs a batch of  PartiQL INSERT statements to add multiple movies to the
```

Comment:
Remove one of the spaces after 'AddMovieBatch runs a batch of  '

Reply:
fixed